frontend/javascripts/types/api_flow_types.ts

```diff
@@ -240,15 +241,15 @@ export type APIDatasetCompactWithoutStatusAndLayerNames = Pick<
   | "name"
   | "folderId"
   | "isActive"
-  | "displayName"
+  | "directoryName"
   | "created"
   | "isEditable"
   | "lastUsedByUser"
   | "tags"
   | "isUnreported"
 >;
 export type APIDatasetCompact = APIDatasetCompactWithoutStatusAndLayerNames & {
-  id?: string;
+  id: string; // Open question: Why was this optional?, The backend code clearly always returns an id ... :thinking:
```

Comment:
Could be legacy or an oversight, but I don't know. I think it's fine to change if typescript doesn't complain.